src/database.rs

```diff
@@ -211,8 +211,15 @@ pub struct MetricsDiff {
     changed_nonascii: bool,
 }
 
+#[derive(Debug, PartialEq)]
+pub enum DiffCode {
+    NoChanges = 0,
+    Changes = 1,
+    Suspicious = 2,
+}
+
 impl EntryDiff {
-    fn show_diff(&self, path: &PathBuf, depth: usize) {
+    fn show_diff(&self, path: &PathBuf, depth: usize) -> DiffCode {
```

Comment:
I wonder if it would make sense to have a separate `summarize_diff` method. In theory, we could add a flag to run in "quiet" mode where it doesn't print anything to stdout but only generates a return code. Easier to add that when these are separate functions. Also, I think we can make summarize cleaner than show.